fastapi-app/services/recipe.py

```diff
@@ -0,0 +1,84 @@
+from sqlalchemy import func
+
+from models import Ingredient, Recipe, recipe_ingredient
+
+
+def service_create_new_recipe(recipe, db):
+    ingredients = {}
+    for ingredient_name in recipe['ingredients'].keys():
+        ingredient = db.query(Ingredient) \
+            .filter(Ingredient.name == ingredient_name) \
+            .first()
+
+        if not ingredient:
+            return {'message': f'Error: Ingredient {ingredient_name} does '
+                               f'not exists.'}
+
+        ingredients[ingredient] = recipe['ingredients'][ingredient_name]
+
+    try:
+        new_recipe = Recipe(
```

Comment:
Actually, without the first commit I was't able to connect `Recipe` and` RecipeIngredient` objects, because `Recipe.id ` was't generated until that commit.